### PR TITLE
Remove gratuitous indirection from ShapeHandle

### LINKED ISSUE #295: Remove the Box from the ShapeHandle
Right now, the `ShapeHandle` is [defined](https://github.com/rustsim/ncollide/blob/master/src/shape/shape.rs#L164) as a `Arc<Box<Shape<N>>`. It would be more efficient to avoid the double allocation by defining it as `Arc<Shape<N>>` instead. This is not possible currently because we need to use `Arc::make_mut` at [some point](https://github.com/rustsim/ncollide/blob/master/src/shape/shape.rs#L179), which relies in the fact that `Box<Shape<N>>` implements `Clone`.

@Ralith suggested that we use [Arc::get_mut](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.get_mut) instead, after making sure we have a unique reference. This has the specificity of working on an `Arc<Shape<N>>`.

### PR DESCRIPTION
**Pull request by Ralith**
Fixes #295

### CODE REVIEW

File: src/shape/shape.rs [hunk 1]
```diff
@@ -9,14 +9,13 @@ use std::sync::Arc;
 use downcast_rs::Downcast;
 
 pub trait ShapeClone<N: RealField> {
-    fn clone_box(&self) -> Box<dyn Shape<N>> {
-        unimplemented!()
-    }
```
Comment (Ralith) [inline src/shape/shape.rs]: I'm not sure why this was here. If people really want a `Shape` that panics on clone, that probably should be something they opt into?

**Comment by Ralith** [comment]
CI failure looks spurious.

File: src/shape/shape.rs (hunk 1 above)
Comment (sebcrozet) [reply src/shape/shape.rs]: Yeah, I forgot to remove this.

**Comment by sebcrozet** [comment]
Thanks!

**Pull request merged**